source/Mlos.Python/mlos/Spaces/HypergridAdapters/ContinuousToPolynomialBasisHypergridAdapter.py

```diff
@@ -0,0 +1,167 @@
+            'order': 'C'
+        }
+        self._target: Hypergrid = None
+
+        if self._adaptee.is_hierarchical():
+            self._adaptee = HierarchicalToFlatHypergridAdapter(adaptee=self._adaptee)
+
+        # Record which adaptee dimensions are continuous
+        self._adaptee_contains_dimensions_to_transform = False
+        self._adaptee_dimension_names_to_transform = []
+        for adaptee_dimension in self._adaptee.dimensions:
+            if isinstance(adaptee_dimension, ContinuousDimension):
+                self._adaptee_dimension_names_to_transform.append(adaptee_dimension.name)
+        self._num_dimensions_to_transform = len(self._adaptee_dimension_names_to_transform)
+        self._adaptee_contains_dimensions_to_transform = self._num_dimensions_to_transform > 0
+
+        # instantiate sklearn's polynomial features instance
+        self._polynomial_features = PolynomialFeatures(**self._polynomial_features_kwargs)
+        # because the exact number of additional dimensions that will be added depends on the parameters to sklearn's PF,
+        # *and* the sklearn PF instance above doesn't determine this information until after the .fit() method is called (requiring a dataframe),
+        # *and* the target hypergrid can not be constructed without knowing the resulting number of continuous dimensions,
+        # a trivial dataframe is constructed (all 1s) and .fit_transform() of _polynomial_features instance is called.
+        trivial_continuous_dim_x = np.array(np.ones((1, self._num_dimensions_to_transform))).reshape(1, -1)
+        trivial_polynomial_features_y = self._polynomial_features.fit_transform(trivial_continuous_dim_x)
+        self._polynomial_features_powers = self._polynomial_features.powers_
+        self._num_polynomial_basis_dimensions_in_target = trivial_polynomial_features_y.shape[1]
+        self._target_polynomial_feature_map = {}  # keys are target dimension names, values are index in features
+        self._build_simple_hypergrid_target()
+
+    def _build_simple_hypergrid_target(self) -> None:
+        self._target = SimpleHypergrid(
+            name=self._adaptee.name,
+            dimensions=None,
+            random_state=self._adaptee.random_state
+        )
+
+        # since linear terms will always be included in the polynomial basis functions, add all dimensions
+        for adaptee_dimension in self._adaptee.dimensions:
+            self._target.add_dimension(adaptee_dimension.copy())
+
+        if not self._adaptee_contains_dimensions_to_transform:
+            return
+
+        # add new dimensions to be created by sklearn PolynomialFeatures
+
+        # construct target dim names using adaptee dim names and polynomial feature powers matrix
+        # The polynomial feature "feature names" look like: ['1', 'x0', 'x1', 'x0^2', 'x0 x1', 'x1^2']
+        # and this is independent of the adaptee dimension names (since the sklearn class doesn't take the x feature names as inputs).

```

Comment:
When Adam and I spoke about the derived dimension names, I heard concern about including white space or potentially reserved characters in these names.  So wanted to have the logic defined to perform the substitutions ourselves.  As it is written, we're using '*', '^', ' ' since that's what PolynomialFeatures uses, but this can be changed easily enough now.  Thanks for pointing out that get_feature_names accepts the original field names though.  Great to know for the future.